src/dask_awkward/lib/core.py

```diff
@@ -557,6 +557,10 @@ def reset_meta(self) -> None:
 
     def repartition(self, npartitions=None, divisions=None, rows_per_partition=None):
         from dask_awkward.lib.structure import repartition_layer
+        from dask.highlevelgraph import MaterializedLayer
+
+        token = tokenize(self, npartitions, divisions, rows_per_partition)
```

Comment:
(I would argue that any mix of npartitions/rows-per/divisions that ends up with the same divisions a few lines down amounts to exactly the same thing and should have the same token)

Reply:
makes sense; just changed this